in_toto/runlib.py

```diff
@@ -264,7 +293,23 @@ def record_artifacts_as_dict(artifacts, exclude_patterns=None,
           # FIXME: this is necessary to provide consisency between windows filepaths and
           # *nix filepaths. A better solution may be in order though...
           normalized_filepath = filepath.replace("\\", "/")
-          artifacts_dict[normalized_filepath] = _hash_artifact(filepath,
+
+          key = normalized_filepath
+          if lstrip_paths:
+            # If a prefix is passed using the argument --lstrip-paths,
+            # that prefix is left stripped from the filepath passed.
+            # Note: if the prefix doesn't include a trailing /, the dictionary key
+            # may include an unexpected /.
+            for prefix in lstrip_paths:
+              if normalized_filepath.startswith(prefix):
+                key = normalized_filepath[len(prefix):]
+                break
+
+          if key in artifacts_dict:
+            raise in_toto.exceptions.PrefixError("Prefix selection has resulted"
+                " in non unique dictionary key '{}'".format(key))
+
+          artifacts_dict[key] = _hash_artifact(filepath,
```

Comment:
Looks like there is potential for DRY here. Mind creating a helper function?

Reply:
For the PrefixError check? It happens twice only (for now), but I suppose it could be broken off.